examples/move/transfer-to-object/README.md

```diff
@@ -0,0 +1,219 @@
+
+Let's go through this implementation in more detail and compare it to the above
+two implementations.
+
+### Interaction Stability: Object ID Remains the Same
+
+To make a payment, nothing changes from Implementation 1. In particular,
+customers will still use `identified_payment::make_payment` and simply set the
+address they want to send to be the object ID of the restaurant's `Register`
+object. If the restaurant changes the ownership of the `Register` object, this
+will be opaque to the customers – they will always send their payment to the
+same `Register` object.
+
+### Receiving Payments
+
+At a high level, handling payments after they have been made using
+transfer-to-object resides somewhere between both Implementation 1 and
+Implementation 2. In particular:
+
+- Similar to Implementation 1, the object IDs of the payments you want to
+  handle in that transaction will show up in the transaction's inputs.
+- Similar to Implementation 2, there are dynamic checks enforced on being able
+  to access the sent payments.
+
+To understand what's going on here, it's best to go through the implementation
+of `handle_payment`:
+
+```move
+/// We take the `Register` shared object mutably, along with a "ticket"
+// `handle_payment` that we can exchange for the actual `IdentifiedPayment` object
+// that it is associated with.
+public fun handle_payment(register: &mut Register, handle_payment: Receiving<IdentifiedPayment>, ctx: &TxContext): IdentifiedPayment {
+    // If the sender of the transaction that wants to handle this payment is in the list of authorized employees in the `Register` object
+    // then we will permit them to withdraw the `IdentifiedPayment` object.
+    assert!(vector::contains(&register.authorized_employees, tx_context::sender(ctx)), ENotAuthorized);
+    // Authorization check successful -- exchange the `handle_payment` ticket
+    // for the `IdentifiedPayment` object and return it.
+    transfer::public_receive(&mut register.id, handle_payment)
+}
+```
+
+### Adding Tips Using a Custom `Receive` Rule
+
+One additional benefit of transfer-to-object is that, in addition to being able
+to specify custom transfer rules for `key`-only objects, you can also specify
+custom receiving rules for `key`-only objects in a very similar manner: if an
+object is `key`-only, then the `sui::transfer::receive` function can be called
+in the module that defines the object, but not elsewhere. Elsewhere, the
+`sui::transfer::public_receive` function must be called and can only be used on
+objects that also have the `store` ability.
+
+With this information, we can define a wrapper around `IdentifiedPayment`s
+where we can earmark that payment for a specific address, e.g., the address of
+our server at the restaurant. We can then use the custom receive rule to ensure
+that only our server can access their tip and no one else can.
+
+```move
+struct EarmarkedPayment has key {
+    id: UID,
+    payment: IdentifiedPayment,
```

Comment:
We can't do this because `IdentifiedPayment` doesn't have `store`, right?

Reply:
(I see now in the actual example `IdentifiedPayment` does have `store`, but in the README it does not).

Reply:
Correct -- I had updated the code so `IdentifiedPayment` had `store` as well but forgot to update it here. So great catch!